polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala

```diff
@@ -246,15 +246,21 @@ class SocketTransport(
 
   private[polynote] def deployAndServe(): RIO[BaseEnv with GlobalEnv with CurrentNotebook with TaskManager, (TransportServer[InetSocketAddress], SocketTransport.DeployedProcess)] =
     TaskManager.run("RemoteKernel", "Remote kernel", "Starting remote kernel") {
-      for {
-        socketServer  <- openServerChannel
-        serverAddress  = socketServer.getLocalAddress.asInstanceOf[InetSocketAddress]
-        process       <- deploy.deployKernel(this, serverAddress)
-        _             <- CurrentTask.update(_.progress(0.5, Some("Waiting for remote kernel")))
-        connection    <- startConnection(socketServer).raceFirst(monitorProcess(process))
-        connection2   <- startConnection(socketServer)
-        server        <- SocketTransportServer(socketServer, connection, connection2, process)
-      } yield (server, process)
+      openServerChannel.flatMap {
```

Comment:
What is the effect of this change? Just rearranging things or is there a logic change I'm not spotting?

Reply:
it was rearranged to allow the `connect` part to have `tapError(_ => process.kill())`

Reply:
aha! does that prevent some of the "zombie kernels" we've seen?